proof/invariant-abstract/Invariants_AI.thy

```diff
@@ -1285,6 +1285,8 @@ lemma valid_objsE [elim]:
   "\<lbrakk> valid_objs s; kheap s x = Some obj; valid_obj x obj s \<Longrightarrow> R \<rbrakk> \<Longrightarrow> R"
   unfolding valid_objs_def by (auto simp: dom_def)
 
+\<comment> \<open>This would be a simp rule but too many existing proofs expect valid_obj and break when it is
+    preemptively simplified into arch_valid_obj.\<close>
```

Comment:
I had the same issue with `capMasterCap` when arch-splitting, it simplified for the "generic" arch object case but then nothing else would work with it in the generic context. I expect this kind of thing is not uncommon.

Reply:
For proofs that involve things like `valid_obj X Y = valid_obj X Y'` or something, I'm not sure whether we'd ever want these as simp, otherwise you'd need to simp del them to deal with the generic concept of valid_obj. Hmm. Not sure about this comment in that case.